### PR TITLE
Fix and silence some Mypy failures

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
### Description

This PR is trying to fix the failing pre-commit.ci CI job on recent PRs. I've fixed two comments and silenced some failures in `packages/netcdf4/test_netcdf4.py` by temporarily ignoring the file because there are many of them right now. I am not sure when or how they came up in the first place, since the file has been untouched for a while. dPlease let me know if you'd like me to fix them here, @hoodmane. Thanks!

### Checklists

N/A

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ repos:
         name: mypy-tests
         args: [--ignore-missing-imports]
         files: ^(packages/|docs|/conftest.py|src/tests)
-        exclude: (^packages/.*/setup.py|/src|^packages/aiohttp/aiohttp_patch.py$|^packages/zfpy/test_zfpy.py$)
+        exclude: (^packages/.*/setup.py|/src|^packages/aiohttp/aiohttp_patch.py$|^packages/zfpy/test_zfpy.py$|^packages/netcdf4/test_netcdf4.py$)
```
Comment (hoodmane) [inline .pre-commit-config.yaml:85]: I wonder if it's possible to exclude files by putting a pragma at the top instead of listing them here.

**Review by hoodmane — APPROVED** [review]
Thanks @agriyakhetarpal !

File: .pre-commit-config.yaml (hunk 1 above)
Comment (agriyakhetarpal) [reply .pre-commit-config.yaml:85]: Ah, there is: https://mypy.readthedocs.io/en/stable/common_issues.html#ignoring-a-whole-file, will implement this change, except for the glob patterns.

Comment (hoodmane) [reply .pre-commit-config.yaml]: Thanks!

**Comment by agriyakhetarpal** [comment]
Thanks for the review!

**Pull request merged**